linkerd/http/prom/src/record_response.rs

```diff
@@ -76,7 +58,7 @@ pub trait StreamLabel: Send + 'static {
         + Sync
         + 'static;
 
-    fn init_response<B>(&mut self, rsp: &http::Response<B>);
+    fn init_response(&mut self, rsp: &http::response::Parts);
```

Comment:
this is a core insight i want to highlight.

we don't need or want to see the response body in our implementations of this hook. we _do_ want to e.g. record the start time, but for now, using `Parts` gives us most of the same information without having to lug around that darn `B` type parameter.

Reply:
ditto for `MkStreamLabel`. inspecting `Parts` means that we can have object-safe versions of these traits, crucial if we want to lean on type erasure and box these up over in our application code.

Reply:
This is awesome! Getting rid of the parameter simplified a lot by the looks of it.